integrations/qdrant/tests/test_retriever.py

```diff
@@ -114,6 +118,11 @@ def test_run(self, filterable_docs: List[Document]):
         for document in results:
             assert document.embedding is None
 
+        results = retriever.run(
+            query_embedding=_random_embeddings(768), top_k=5, return_embedding=False, score_threshold=0.999
+        )["documents"]
+        assert len(results) == 0
+
```

Comment:
Let's introduce a specific test (`run_with_score_threshold`).
I would also suggest using a lower `score_threshold` and check it is respected by the returned docs.

Reply:
I started with that, and realized that the random embeddings lead to changing scores. Perhaps adding a random seed might be helpful here?

Reply:
You can create 2 or 3 example documents with fixed embedding.
[Here is an example in Weaviate tests](https://github.com/deepset-ai/haystack-core-integrations/blob/f73c3514c13cdaaf10a878e88f5987529958e9d2/integrations/weaviate/tests/test_document_store.py#L579).